mail/protocols/smtp/src/test/java/com/fsck/k9/mail/transport/smtp/SmtpTransportTest.kt

```diff
@@ -459,12 +447,9 @@ class SmtpTransportTest {
         }
         val transport = startServerAndCreateSmtpTransport(server, authenticationType = AuthType.EXTERNAL)
 
-        try {
-            transport.open()
-            fail("Exception expected")
-        } catch (e: CertificateValidationException) {
-            assertThat(e.reason).isEqualTo(CertificateValidationException.Reason.MissingCapability)
-        }
+        assertThat { transport.open() }.isFailure()
+            .isInstanceOf(CertificateValidationException::class)
+            .transform { it.reason }.isEqualTo(CertificateValidationException.Reason.MissingCapability)
```

Comment:
Prop could be used if you call any method without arguments. Transform when you need to provide arguments.

But when the underlying class is written in Java it can't use Kotlin propery access syntax due to: `Unsupported [reference to the synthetic extension property for a Java get/set method]`.  So `transform` avoids this error.

But I just found a workaround for `prop`: use  `.prop(CertificateValidationException::getReason)` instead.